x-pack/plugins/enterprise_search/public/applications/app_search/components/settings/log_retention/log_retention_logic.test.ts

```diff
@@ -0,0 +1,370 @@
+              enabled: true,
+            },
+          }),
+        });
+
+        await promise;
+        expect(LogRetentionLogic.actions.updateLogRetention).toHaveBeenCalledWith(
+          TYPICAL_CLIENT_LOG_RETENTION
+        );
+
+        expect(LogRetentionLogic.actions.clearLogRetentionUpdating).toHaveBeenCalled();
+      });
+
+      it('handles errors', async () => {
+        const promise = Promise.reject('An error occured');
+        http.put.mockReturnValue(promise);
+
+        LogRetentionLogic.actions.saveLogRetention(ELogRetentionOptions.Analytics, true);
+
+        try {
+          await promise;
+        } catch {
+          // Do nothing
+        }
+        expect(flashAPIErrors).toHaveBeenCalledWith('An error occured');
+        expect(LogRetentionLogic.actions.clearLogRetentionUpdating).toHaveBeenCalled();
+      });
+    });
+
+    describe('toggleLogRetention', () => {
+      describe('logsRetentionUpdating', () => {
+        it('sets logsRetentionUpdating to true', () => {
+          mount({
+            logsRetentionUpdating: false,
+          });
+
+          LogRetentionLogic.actions.toggleLogRetention(ELogRetentionOptions.Analytics);
+
+          expect(LogRetentionLogic.values).toEqual({
+            ...DEFAULT_VALUES,
+            logsRetentionUpdating: true,
+          });
+        });
+      });
+
+      it('will call setOpenModal if already enabled', () => {
+        mount({
+          logRetention: {
+            [ELogRetentionOptions.Analytics]: {
+              enabled: true,
+            },
+          },
+        });
+        jest.spyOn(LogRetentionLogic.actions, 'setOpenModal');
+
+        LogRetentionLogic.actions.toggleLogRetention(ELogRetentionOptions.Analytics);
+
+        expect(LogRetentionLogic.actions.setOpenModal).toHaveBeenCalledWith(
+          ELogRetentionOptions.Analytics
+        );
+      });
+    });
+
+    it('will call saveLogRetention if NOT already enabled', () => {
+      mount({
+        logRetention: {
+          [ELogRetentionOptions.Analytics]: {
+            enabled: false,
+          },
+        },
+      });
+      jest.spyOn(LogRetentionLogic.actions, 'saveLogRetention');
+
+      LogRetentionLogic.actions.toggleLogRetention(ELogRetentionOptions.Analytics);
+
+      expect(LogRetentionLogic.actions.saveLogRetention).toHaveBeenCalledWith(
+        ELogRetentionOptions.Analytics,
+        true
+      );
+    });
+
+    it('will do nothing if logRetention option is not yet set', () => {
+      mount({
+        logRetention: {},
+      });
+      jest.spyOn(LogRetentionLogic.actions, 'saveLogRetention');
+      jest.spyOn(LogRetentionLogic.actions, 'setOpenModal');
+
+      LogRetentionLogic.actions.toggleLogRetention(ELogRetentionOptions.API);
+
+      expect(LogRetentionLogic.actions.saveLogRetention).not.toHaveBeenCalled();
+      expect(LogRetentionLogic.actions.setOpenModal).not.toHaveBeenCalled();
+    });
+  });
+
+  describe('fetchLogRetention', () => {
```

Comment:
One more super quick q - I noticed this wasn't in the `actions` block, was that for a specific reason? Was a little confusing to me as the other 2 listeners are still in the main actions block.

Reply:
No I think I just missed this.